File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
import sys

input = sys.stdin.readline


heights = [int(input()) for _ in range(9)]
heights.sort()


left = 0
right = len(heights) - 1
currSum = sum(heights) - heights[left] - heights[right]

while left < right:
    if currSum == 100:
        break
    elif currSum < 100:
        currSum += heights[right]
        right -= 1
        currSum -= heights[right]
    else:
        currSum += heights[left]
        left += 1
        currSum -= heights[left]

for i in range(len(heights)):
    if i != left and i != right:
        print(heights[i])

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;

public class Main {

  static StringBuilder sb = new StringBuilder();
  static int[] heights;

  public static void main(String[] args) throws IOException {
    setUp();

    solve();

    output();
  }

  private static void solve() {
    Arrays.sort(heights);
    int total = Arrays.stream(heights).sum();
    for (int a = 0; a < 9; a++) {
      for (int b = a + 1; b < 9; b++) {
        if (total - heights[a] - heights[b] == 100) {
          for (int i = 0; i < 9; i++) {
            if (i == a || i == b) {
              continue;
            }
            sb.append(heights[i]).append("\n");
          }

          return;
        }
      }
    }

  }

  private static void setUp() throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    heights = new int[9];
    for (int i = 0; i < 9; i++) {
      heights[i] = Integer.parseInt(br.readLine());
    }

  }

  private static void output() throws IOException {
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    bw.write(sb.toString());
    bw.flush();
    bw.close();
  }

}